### PR TITLE
Reorganize the store management link section and provide extensibility

### LINKED ISSUE #5244: Home Screen - reorganize Store Management section
**Description:** we're going to reorganize the store management areas in different categories, making it easier for users to find the shortcut that they want to access

**Acceptance criteria:**
- [ ] The Marketing & Merchandising category is going to include the following shortcuts: Marketing, Add Products, Personalize my Store, View my Store
- [ ] The Settings category is going to include the following shortcuts: Store Details, Payments, Tax, Shipping
- [ ] The Get Support shortcut is going to be included in the Help panel that's going to be added to the home screen
- [ ] In the new single-column layout, the Store management card should expand to 2 columns.

**Event tracking:**
NA

**Additional links:**
[P2 with links to the Figma file](pbIJXs-ma-p2)

### PR DESCRIPTION
**Pull request by samueljseay**
Fixes #5244 and #5245

This is a refactor of the store management links to reflect the new designs and also provides extensibility, allowing developers to add their own links to this section under an "Extensions" category.

To use the new filter 3PDs will be able to call `addFilter` with the new filter `'woocommerce_admin_homescreen_quicklinks'`. Note that we have explicitly chosen to disable links that link outside of the current host.

Originally the store management links used the `@woocommerce/list` component, but this was no longer suitable because it would have required major redesign of that component and much of that component's functionality didn't feel necessary for this.

Note that the new components I've implemented could be easily promoted to the `@woocommerce/components` package later if we decide they are more broadly useful, but I'd suggest we let this component mature and let the new designs mature before promoting it.

### Accessibility

- [x] I've tested using only a keyboard (no mouse)
- [x] I've tested using a screen reader (Voice over)
- [x] All text has [at least a 4.5 color contrast with its background](https://webaim.org/resources/contrastchecker/)

### Screenshots

There are 2 display modes.

in 2 column mode:

Screenshot 2020-10-27 at 7 39 47 PM

in 1 column mode:

Screenshot 2020-10-27 at 7 40 03 PM

#### Visual/behaviour
1. On a site with the task list completed visit the homescreen
2. Try out all the links provided. 
3. Check that tracking works.
4. Visual inspection against the designs. See pbIJXs-ma-p2
5. Try the different display modes using the new activity panel item "Display"

#### Extensibility

Here's a code snippet for testing extensibility, place it anywhere in our JS code that runs before the store management component is rendered:

```
addFilter(
		'woocommerce_admin_homescreen_quicklinks',
		'my-cool-extension',
		( quickLinks ) => {
			return [
				...quickLinks,
				{
					title: __( 'My Cool link', 'woocommerce-admin' ),
					href: 'foo/bar',
					icon: <div>Hello World</div>,
				},
			];
		}
	);
```

Note that if you want to try a real icon, then you'll need to import it from `@wordpress/icons`. If you'd like to to test the filtering of external links then change `href`'s value to be `https://example.com`.

### Changelog Note:

Dev: Rearrange the store management links under categories and expose extensibility via the `woocommerce_admin_homescreen_quicklinks` filter.

### CODE REVIEW

File: client/stylesheets/shared/_gutenberg-components.scss
Comment (samueljseay) [inline client/stylesheets/shared/_gutenberg-components.scss:26]: Note that this change is borrowed from another PR that fixes issues with wp components. I can remove this before merge, but its going to be easier to test visuals with it.

File: client/store-management-links/index.js
Comment (jeffstieler) [inline client/store-management-links/index.js:204]: Is there a future use for `linkType`, or is it just to determine the `isExternal` prop? If the latter, what do you think about simplifying this to just be `isExternal` on the link objects instead?

File: client/store-management-links/quick-link-category/index.js
Comment (jeffstieler) [inline client/store-management-links/quick-link-category/index.js:4]: Do we need to import `React` here?

File: client/store-management-links/quick-link-category/test/index.js
Comment (jeffstieler) [inline client/store-management-links/quick-link-category/test/index.js:13]: I was expecting to see an assertion for the title being displayed. Did that get missed?

File: client/store-management-links/quick-link-category/test/index.js
Comment (jeffstieler) [inline client/store-management-links/quick-link-category/test/index.js:5]: This should be necessary as it's included in [`tests/js/setup-react-testing-library`](https://github.com/woocommerce/woocommerce-admin/blob/1e0061d0df8c177795c58ab9603af95f4aef6610/tests/js/setup-react-testing-library.js#L5).

File: client/store-management-links/quick-link/index.js
Comment (jeffstieler) [inline client/store-management-links/quick-link/index.js:4]: Same question about necessity.

File: client/store-management-links/quick-link/index.js
Comment (jeffstieler) [inline client/store-management-links/quick-link/index.js:17]: Could we just use `onClick` here instead of defining a closure?

File: client/store-management-links/quick-link/test/index.js
Comment (jeffstieler) [inline client/store-management-links/quick-link/test/index.js:4]: Needed?

File: client/store-management-links/quick-link/test/index.js
Comment (jeffstieler) [inline client/store-management-links/quick-link/test/index.js:7]: Shouldn't be needed.

File: client/store-management-links/quick-link/test/index.js
Comment (jeffstieler) [inline client/store-management-links/quick-link/test/index.js:16]: Is this missing the assertion for the icon?

File: client/store-management-links/quick-link/index.js
Comment (jeffstieler) [inline client/store-management-links/quick-link/index.js:16]: Should this use the `Link` component from `@woocommerce/components`? That would allow for navigating to other wc-admin powered pages without a full page load (if passed the `type` prop). Alternate idea: should we just update the `Link` component to be this `QuickLink` implementation?

**Review by jeffstieler — CHANGES_REQUESTED** [review]
Code review aside, this is testing well but with some console warnings.

File: client/store-management-links/index.js
Comment (samueljseay) [reply client/store-management-links/index.js:154]: oooh good catch. 🤔 I'll refactor this.

File: client/store-management-links/index.js
Comment (samueljseay) [reply client/store-management-links/index.js:204]: @jeffstieler I went back and forth on this and almost removed it. Your validation makes me think we should remove it. Will do.

File: client/store-management-links/quick-link-category/index.js
Comment (samueljseay) [reply client/store-management-links/quick-link-category/index.js:4]: @jeffstieler Yeah, you always have to import React when using jsx.

Soon we should be able to remove this though; https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#how-to-upgrade-to-the-new-jsx-transform

File: client/store-management-links/quick-link-category/test/index.js
Comment (samueljseay) [reply client/store-management-links/quick-link-category/test/index.js:5]: I missed that, thanks!

File: client/store-management-links/quick-link-category/test/index.js
Comment (samueljseay) [reply client/store-management-links/quick-link-category/test/index.js:13]: It sure did! Thanks for spotting that.

File: client/store-management-links/quick-link/index.js
Comment (samueljseay) [reply client/store-management-links/quick-link/index.js:4]: Yeah you gotta import React to use jsx

File: client/store-management-links/quick-link/index.js
Comment (samueljseay) [reply client/store-management-links/quick-link/index.js:16]: @jeffstieler thats a good point. My main concern would be styling but we could probably override it with css. 🤔 

I had forgotten that `Link` worked in that way so it *could* make sense to use that here. I'll investigate this.

File: client/store-management-links/quick-link/index.js
Comment (samueljseay) [reply client/store-management-links/quick-link/index.js:17]: :D yep good spotting

File: client/store-management-links/quick-link/test/index.js
Comment (samueljseay) [reply client/store-management-links/quick-link/test/index.js:7]: I'll remove this.

File: client/store-management-links/quick-link/test/index.js
Comment (samueljseay) [reply client/store-management-links/quick-link/test/index.js:16]: Oh yes it is, I struggled with doing assertions on the icon svg for some reason. I'll remove that from the assertion

File: client/homescreen/test/index.js
Comment (samueljseay) [reply client/homescreen/test/index.js:24]: Good spotting I missed removing this.

File: client/store-management-links/index.js
Comment (samueljseay) [reply client/store-management-links/index.js:211]: @jeffstieler yep `task_name` is correct, this track was originally happening in the QuickLinks component heres the OG code:

```
props.recordEvent( 'home_quick_links_click', {
		task_name: listItemTag,
} );
```

File: client/store-management-links/index.js
Comment (samueljseay) [reply client/store-management-links/index.js]: Removed in favor of `isExternal` here: https://github.com/woocommerce/woocommerce-admin/pull/5476/commits/b96c0605b75369f792246b9b1755c5c9a7265259

File: client/homescreen/test/index.js
Comment (samueljseay) [reply client/homescreen/test/index.js]: Removed in https://github.com/woocommerce/woocommerce-admin/pull/5476/commits/a25ec4fedf6ec7555a9461b22c798f2fdfea161d

File: client/store-management-links/index.js
Comment (samueljseay) [reply client/store-management-links/index.js]: I have opted to go with the `Link` component you suggested in your other comment so have kept the linkType. Changes here: https://github.com/woocommerce/woocommerce-admin/pull/5476/commits/24e2171719d231dcd39305af5ff443a891f509ea

File: client/store-management-links/quick-link/index.js
Comment (samueljseay) [reply client/store-management-links/quick-link/index.js]: This is resolved now when passing to the `Link` component in https://github.com/woocommerce/woocommerce-admin/pull/5476/commits/24e2171719d231dcd39305af5ff443a891f509ea

File: client/store-management-links/quick-link/test/index.js
Comment (samueljseay) [reply client/store-management-links/quick-link/test/index.js]: Removed the assertion. I still haven't had luck querying svgs by role or any other attribute in jest-dom.

File: client/store-management-links/quick-link/index.js
Comment (samueljseay) [reply client/store-management-links/quick-link/index.js]: I've gone with the `Link` component here and it works well. 🎉 thanks this was a definite improvement. https://github.com/woocommerce/woocommerce-admin/pull/5476/commits/24e2171719d231dcd39305af5ff443a891f509ea

File: client/store-management-links/index.js
Comment (samueljseay) [reply client/store-management-links/index.js:154]: @jeffstieler This should be fixed in https://github.com/woocommerce/woocommerce-admin/pull/5476/commits/90de31dab679d216c2f742f38d2dffd3559d88fa I've opted to create a stable key for the item based on a combo of title/tag/href and the only way it'll collide is if devs provide duplicate links which probably isn't a bad thing for them to be warned about.

File: client/stylesheets/shared/_gutenberg-components.scss
Comment (samueljseay) [reply client/stylesheets/shared/_gutenberg-components.scss:26]: Just a note, I'm going to remove this now and remerge `main` because we have a fix for this.

File: client/store-management-links/quick-link/index.js
Comment (jeffstieler) [reply client/store-management-links/quick-link/index.js:4]: Curious.. most all of our other JSX files don't import React.

File: client/store-management-links/quick-link/index.js
Comment (jeffstieler) [inline client/store-management-links/quick-link/index.js:19]: Why isn't the `linkType` prop being passed to `<Link>`?

**Review by jeffstieler — APPROVED** [review]
The code changes look good for the most part, but I do have a question about the `linkType` prop omission from the `<Link>` component (it defaults to `wc-admin` which really only fits for one quick link at the moment). Also - perhaps for a follow up PR - are we going to support the new "extension" `linkType` or perhaps let extensions choose their own `linkType`?

Pre-approving, assuming there's agreement on the `linkType` prop being passed to `<Link>`s.

File: client/store-management-links/quick-link/index.js
Comment (samueljseay) [reply client/store-management-links/quick-link/index.js:19]: Thanks for spotting this, just an error of omission. Will fix.

Comment (jeffstieler) [reply client/store-management-links/quick-link/index.js:19]: I think for the links we're creating we should explicitly pass a `linkType` of "wp-admin" or "wc-admin". I doubt there's much to break with the regex method for detecting if an external/extension link could actually be a "wc-admin" type, but out of caution how about we force all external/extension links to be of type "wp-admin"?

Comment (samueljseay) [reply client/store-management-links/quick-link/index.js:19]: Sounds good. I'll ship that

Comment (samueljseay) [reply client/store-management-links/quick-link/index.js:19]: @jeffstieler I had another think about this and I want to break down my reasoning about the approach I'll take here based on your feedback.

There are several link types generated by the original quick links code:

`wc-settings`, `wc-admin`, `wp-admin` and `external`.

There are fewer link types respected by the `<Link>` component:

`wp-admin`, `wc-admin`, `external`

I think the right choice here is to ignore the fact that `wp-admin` and `wc-admin` links are not functionally different. Rather we should respect the API in case someone else decides that perhaps external links should behave differently for example.

So steps I'll take are:

1. Remove the `wc-settings` type from existing code. Replace it with `wp-admin` type
2. Normalise all `extension` links to `wp-admin` type
3. For remaining `wc-admin` links I will keep them as `wc-admin` type and pass the link type down to `<Link>`

I think this is similar to what you suggested but your suggestions were a little ambiguous so I've spelled it out here.

Comment (samueljseay) [reply client/store-management-links/quick-link/index.js:19]: @jeffstieler I had another think about this and I want to break down my reasoning about the approach I'll take here based on your feedback.

The link types we normalise to pass down to `<QuickLink>` right now include:

`wc-admin`, `wp-admin`, `external` and `extension`

There are fewer link types respected by the `<Link>` component:

`wp-admin`, `wc-admin`, `external`

I think the right choice here is to ignore the fact that `wp-admin` and `wc-admin` links are not functionally different. Rather we should respect the API in case someone else decides that perhaps external links should behave differently for example.

So steps I'll take are:

1. Remove the `extension` type from existing code. Replace it with `wp-admin` type
2. For remaining `wc-admin` links I will keep them as `wc-admin` type and pass the link type down to `<Link>`

I think this is similar to what you suggested but it felt a little ambiguous so I've spelled it out here.

Comment (samueljseay) [reply client/store-management-links/quick-link/index.js:19]: I've addressed this here: https://github.com/woocommerce/woocommerce-admin/pull/5476/commits/5b5af412fa8b47224f8d804cf1e3289f0b584a40

I'm going to merge this based on your pre-approval, but if you see any problems with the approach I've taken here then please let me know and I can address it in a follow up PR.

**Pull request merged**
